spring-integration-core/src/test/java/org/springframework/integration/channel/DatatypeChannelTests.java

```diff
@@ -77,14 +79,15 @@ public void unsupportedTypeButConversionServiceSupports() {
 		assertThat(channel.send(new GenericMessage<String>("123"))).isTrue();
 	}
 
-	@Test(expected = MessageDeliveryException.class)
+	@Test
 	public void unsupportedTypeAndConversionServiceDoesNotSupport() {
 		QueueChannel channel = createChannel(Integer.class);
 		ConversionService conversionService = new DefaultConversionService();
 		DefaultDatatypeChannelMessageConverter converter = new DefaultDatatypeChannelMessageConverter();
 		converter.setConversionService(conversionService);
 		channel.setMessageConverter(converter);
-		assertThat(channel.send(new GenericMessage<Boolean>(Boolean.TRUE))).isTrue();
+		assertThatExceptionOfType(MessageDeliveryException.class)
+				.isThrownBy(() -> assertThat(channel.send(new GenericMessage<Boolean>(Boolean.TRUE))).isTrue());
```

Comment:
The assertion in the lambda does not make sense.
If we expect a `MessageDeliveryException`, then `send` is never going to be `true`.
It is just suppressed by that exception:
```suggestion
				.isThrownBy(() -> channel.send(new GenericMessage<Boolean>(Boolean.TRUE)));
```